File: problem1-5/problem2/problem2.py
#!/usr/bin/env python
# By considering the terms in the Fib sequence whose
# values do not exceed four million, find the sum of
# the even-valued terms.

def soln(n=4000000):
    ans = 0
    fib_ini = 1
    fib_dos = 2
    while fib_dos < n:
        if fib_dos % 2 == 0:
            ans += fib_dos
        fib_tres = fib_ini + fib_dos
        fib_ini = fib_dos
        fib_dos = fib_tres
    print("Solution: {0}".format(ans))

if __name__ == "__main__":
    soln()

File: problem1-5/problem2/problem2.js
#!/usr/bin/env js

// By considering the terms in the Fibonacci sequence whose
// values do not exceed four million, find the sum of the
// even-valued terms.

function fourMillionEvenFibs(){
    var ans = 0;
    var fib_plus_two = 2;
    var fib_plus_one = 1;
    while (fib_plus_two < 4000000) {
        if (fib_plus_two % 2 === 0) {
            ans += fib_plus_two;
        }
        var n_plus_one = fib_plus_one + fib_plus_two;
        fib_plus_one = fib_plus_two;
        fib_plus_two = n_plus_one;
    }
    return ans;
}
console.log("Solution: " + fourMillionEvenFibs());
